unblob/processing.py

```diff
@@ -119,3 +149,49 @@ def calculate_unknown_chunks(
         unknown_chunks.append(unknown_chunk)
 
     return unknown_chunks
+
+
+def calculate_entropy(path: Path, *, draw_plot: bool):
+    """Calculate and log shannon entropy divided by 8 for the file in 1mB chunks."""
+    logger.info("Calculating entropy for file", path=path)
+    percentages = []
+
+    # We could use the chunk size instead of another syscall,
+    # but we rely on the actual file size written to the disk
+    file_size = path.stat().st_size
+
+    # Smaller chuk size would be very slow to calculate. This takes ~ 3sec for a 4,5 GB file.
+    buffer_size = 1024 * 1024
+
+    with path.open("rb") as file:
+        for chunk in iterate_file(file, 0, file_size, buffer_size=buffer_size):
+            entropy = shannon_entropy(chunk)
+            entropy_percentage = round(entropy / 8 * 100, 2)
+            percentages.append(entropy_percentage)
+
+    logger.info(
+        "Entropy calculated",
+        mean=statistics.mean(percentages),
+        highest=max(percentages),
+        lowest=min(percentages),
+    )
+
+    if draw_plot:
+        draw_entropy_plot(percentages)
+
+
+def draw_entropy_plot(percentages: List[float]):
+    plt.clear_data()
+    plt.colorless()
+    plt.title("Entropy distribution")
+    plt.xlabel("mB")
+    plt.ylabel("entropy %")
+
+    plt.scatter(percentages, marker="dot")
+    # 16 height leaves no gapes between the lines
+    plt.plot_size(80, 16)
+    plt.ylim(0, 100)
+    # Always show 0% and 100%
+    plt.yticks(range(0, 101, 10))
+
+    logger.debug("Entropy chart", chart=plt.build())
```

Comment:
The log looks a little bit awkward:
```
2022-01-26 11:36.32 [debug    ] Entropy chart                  chart=                              Entropy distribution  
<plot>
```

I think if you use `chart=plt.build().lstrip())` instead, it will remove the spaces before the "Entropy distribution" text to look better.

Reply:
Either that or insert a newline at the beginning. The "Entropy distribution" string is the title that goes on top of the chart, right in the middle (hence the whitespaces) :)